### PR TITLE
Add service store con snippet and editor support

### PR DESCRIPTION
**Pull request by hardikmaheshwari**
## Summary
Fixing a new packageable connection editor to list all available connection types. 
Adding support for service store connection editor and service store connection snippet suggestion

## How did you test this change?

- [ ] Test(s) added
- [X] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

[![Snippet Working]](https://user-images.githubusercontent.com/19693874/207277352-86c3f3a2-18af-4f8b-bc97-7a68788ef065.mov)

[![Editor Working]](https://user-images.githubusercontent.com/19693874/207277489-04ea683c-0fe9-44e0-829b-624e46886366.mov)

### CODE REVIEW

File: packages/legend-application-studio/src/components/editor/side-bar/CreateNewElementModal.tsx [hunk 1]
```diff
@@ -198,27 +201,8 @@ const NewPureModelConnectionDriverEditor = observer(
     newConnectionDriver: NewPackageableConnectionDriver;
     newConnectionValueDriver: NewPureModelConnectionDriver;
   }) => {
-    const { newConnectionDriver, newConnectionValueDriver } = props;
+    const newConnectionValueDriver = props.newConnectionValueDriver;
```
Comment (gayathrir11) [inline packages/legend-application-studio/src/components/editor/side-bar/CreateNewElementModal.tsx:204]: why did you remove this line?

File: packages/legend-application-studio/src/stores/DSL_Mapping_LegendStudioApplicationPlugin_Extension.ts [hunk 2]
```diff
@@ -86,9 +87,16 @@ export type RuntimeConnectionTooltipTextBuilder = (
 
 export type NewConnectionDriverCreator = (
   editorStore: EditorStore,
-  store: Store,
+  typeOrStore: Store | string,
```
Comment (gayathrir11) [inline packages/legend-application-studio/src/stores/DSL_Mapping_LegendStudioApplicationPlugin_Extension.ts:90]: Can we rename this to something else?

File: packages/legend-application-studio/src/components/editor/side-bar/CreateNewElementModal.tsx [hunk 3]
```diff
@@ -302,17 +274,50 @@ const NewConnectionDriverEditor = observer(() => {
...
+  const extraOptionTypes = editorStore.pluginManager
+    .getApplicationPlugins()
+    .flatMap(
+      (plugin) =>
+        (
+          plugin as DSL_Mapping_LegendStudioApplicationPlugin_Extension
+        ).getExtraConnectionTypeOptions?.() ?? [],
+    );
+  const connectionOptions = [
+    PURE_MODEL_CONNECTION,
```
Comment (gayathrir11) [inline packages/legend-application-studio/src/components/editor/side-bar/CreateNewElementModal.tsx:286]: Let's have the CONNECTION_TYPE enum and build everything from that.

File: packages/legend-application-studio/src/stores/editor/NewElementState.ts [hunk 4]
```diff
@@ -317,22 +334,8 @@ export class NewRelationalDatabaseConnectionDriver extends NewConnectionValueDri
   }
 }
 
-export enum CONNECTION_TYPE {
-  RELATIONAL = 'RELATIONAL',
```
Comment (gayathrir11) [inline packages/legend-application-studio/src/stores/editor/NewElementState.ts:321]: Let's not delete this enum

File: .changeset/loud-worms-teach.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+---
+'@finos/legend-application-studio': patch
+'@finos/legend-extension-store-service-store': patch
+---
+
```
Comment (gayathrir11) [inline .changeset/loud-worms-teach.md:5]: Please split the changeset

File: packages/legend-application-studio/src/components/editor/side-bar/CreateNewElementModal.tsx (hunk 1 above)
Comment (hardikmaheshwari) [reply packages/legend-application-studio/src/components/editor/side-bar/CreateNewElementModal.tsx]: Thanks done

File: packages/legend-application-studio/src/stores/DSL_Mapping_LegendStudioApplicationPlugin_Extension.ts (hunk 2 above)
Comment (hardikmaheshwari) [reply packages/legend-application-studio/src/stores/DSL_Mapping_LegendStudioApplicationPlugin_Extension.ts:90]: what would you suggest, not sure I understand the issue with this

File: packages/legend-application-studio/src/components/editor/side-bar/CreateNewElementModal.tsx (hunk 3 above)
Comment (hardikmaheshwari) [reply packages/legend-application-studio/src/components/editor/side-bar/CreateNewElementModal.tsx]: Thanks done

File: packages/legend-application-studio/src/stores/editor/NewElementState.ts (hunk 4 above)
Comment (hardikmaheshwari) [reply packages/legend-application-studio/src/stores/editor/NewElementState.ts:321]: Thanks done

File: .changeset/loud-worms-teach.md (hunk 5 above)
Comment (hardikmaheshwari) [reply .changeset/loud-worms-teach.md:4]: Thanks done

**Review by MauricioUyaguari — APPROVED** [review]

**Comment by MauricioUyaguari** [comment]
@hardikmaheshwari @gayathrir11  have approved and merged but i think overall the new connection modal needs work (before this MR). 
Its a bit confusing if the store selection changes options. 
I think the connection type should be the main driver of all the selections. 
Have created this issue, feel free to add your thoughts https://github.com/finos/legend-studio/issues/1779
